MatrixSDK/Crypto/Algorithms/Megolm/MXMegolmDecryption.m

```diff
@@ -114,7 +115,10 @@ - (MXEventDecryptionResult *)decryptEvent:(MXEvent*)event inTimeline:(NSString*)
         }
         else if ([olmError.domain isEqualToString:MXDecryptingErrorDomain] && olmError.code == MXDecryptingErrorUnknownInboundSessionIdCode)
         {
-            [self addEventToPendingList:event inTimeline:timeline];
+            // Do nothing more on the calling thread
+            dispatch_async(crypto.cryptoQueue, ^{
+                [self addEventToPendingList:event inTimeline:timeline];
+            });
```

Comment:
There is another call to addEventToPendingList in line 105. It can be a good idea to move it to cryptoQueue also. Also asserting thread in addEventToPendingList method would make us sure.